File: stock span.cpp
// stock span 


#include<bits/stdc++.h>
using namespace std;

int main(){
	
	vector<int>  inp ={100,80,60,70,60,75,85};
	
	vector<int> ngeL(inp.size());
	
	int n=inp.size();
	
	ngeL[0]=1;
	
	stack<int> s;
	
	s.push(0);
	
	for(int i=1;i<n;i++){
		
		while(!s.empty() && inp[i]>inp[s.top()]){
			
			s.pop();
			
		}
		
		if(s.empty()){
			ngeL[i]=i+1;
		}else
		{
		ngeL[i]=i - s.top();	
		}
		s.push(i);
	}
	
	
	for(int i=0;i<ngeL.size();i++){
		
		cout<<ngeL[i]<<" ";
		
	}
	
}

File: src/Daily/StockSpan.java
package Daily;

import java.util.Stack;

public class StockSpan {
    public static void main(String[] args){
        int[] arr={};
        int[] span=new int[arr.length];
        Stack<Integer> s=new Stack();
        s.push(0);
        for(int i=1;i<span.length;i++){
            while(!s.empty() && s.peek() <arr[i] ){
                s.pop();
            }
            if(s.size()==0){
                span[i]=i+1;
            }else
            {
                span[i]=i-s.peek();
            }
            s.push(i);
        }
        for(int i=0;i<span.length;i++){
            System.out.println(span[i]);
        }
    }
}
